src/Command/Issue/LabelIssuesCommand.php

```diff
@@ -114,50 +132,67 @@ protected function execute(InputInterface $input, OutputInterface $output)
-            $this->showLabels($output, $labelNames);
-
-            $validation = function ($label) use ($labelNames) {
-                $labels = explode(',', $label);
-                foreach ($labels as $item) {
-                    if (!in_array($item, array_values($labelNames))) {
-                        throw new \InvalidArgumentException(sprintf('Label "%s" is invalid.', $item));
-                    }
-                }
-
-                return $label;
-            };
-
-            /** @var QuestionHelper $question */
-            $questionHelper = $this->getHelper('question');
-            if (!$label = $input->getOption('label')) {
-                $label = $questionHelper->ask(
-                    $input,
-                    $output,
-                    (new Question('<comment>Label(s)?</comment> '))
-                        ->setValidator($validation)
-                        ->setAutocompleterValues($labelNames)
-                );
-            }
+            $styleHelper->writeln(sprintf($issueTitleFormat, $issue['number'], $issue['title']));
+            $styleHelper->newLine();
 
-            // updates the issue
-            $tracker->updateIssue($issue['number'], ['labels' => explode(',', $label)]);
+            $styleHelper->writeln(' <info>Current labels: </info>');
+            $this->getIssueLabels($issue['labels'], $styleHelper);
+
+            $styleHelper->writeln(' <info>Available labels: </info>');
+            $this->getIssueLabels($labelNames, $styleHelper);
+
+            $labels = $styleHelper->askQuestion(
+                (new Question('<comment>Assign label(s)</comment> ', implode(', ', $issue['labels'])))
+                    ->setValidator($validation)
+                    ->setAutocompleterValues($labelNames)
+            );
+
+            // Sort to ensure they can be equal
+            sort($labels);
+            sort($issue['labels']);
+
+            if ($labels !== $issue['labels']) {
+                $tracker->updateIssue($issue['number'], ['labels' => $labels]);
+                $styleHelper->success(sprintf('Updated issue/pull-request #%d.', $issue));
+            }
         }
 
+        $styleHelper->success('Issues/pull requests are updated.');
+
         return self::COMMAND_SUCCESS;
     }
 
-    private function showLabels(OutputInterface $output, array $labels)
+    private function validateLabels($input, array $acceptedLabels)
     {
-        /** @var TableHelper $table */
-        $table = $this->getHelper('table');
-        $table->setRows(array_chunk($labels, 3));
-        $table->render($output);
+        $inputLabels = array_map('trim', explode(',', $input));
+        $labels = [];
+
+        foreach ($inputLabels as $label) {
+            if ('' === $label) {
+                continue;
+            }
+
+            if (!in_array($label, $acceptedLabels, true)) {
+                throw new \InvalidArgumentException(
```

Comment:
validateLabels() is used as a callback for the question promt, so any exception will be cached already.